src/fastfield/mod.rs

```diff
@@ -62,7 +63,9 @@ pub trait FastValue: Clone + Copy + Send + Sync + PartialOrd + 'static {
     /// Converts a value to u64.
     ///
     /// Internally all fast field values are encoded as u64.
-    fn to_u64(&self) -> u64;
```

Comment:
I understand you use the field type for truncation on datetime, but it seems too convoluted. Let's not change the `to_u64`.

Instead, let's put the truncation at the fast field writer level.